src/ophyd_async/panda/writers/hdf_writer.py

```diff
@@ -0,0 +1,234 @@
+import asyncio
+import atexit
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, AsyncIterator, Dict, List, Optional, Union
+
+from bluesky.protocols import Asset, Descriptor, Hints
+from p4p.client.thread import Context
+
+from ophyd_async.core import (
+    DEFAULT_TIMEOUT,
+    DetectorWriter,
+    Device,
+    DirectoryProvider,
+    NameProvider,
+    SignalR,
+    SignalRW,
+    wait_for_value,
+)
+from ophyd_async.panda.panda import PandA
+
+from .panda_hdf import _HDFDataset, _HDFFile
+
+
+@dataclass()
+class HdfSignals:
+    file_path: SignalRW  # This is the directory rather than path. Path is read only
+    file_name: SignalRW
+    num_capture: SignalRW
+    num_captured: SignalR
+    capture: SignalRW
```

Comment:
That's a different approach to the one I was expecting. I was thinking that you would have something like this in `panda.py`:
```
class DataBlock(Device):
    capture: SignalRW[bool]
    num_capture: SignalRW[int]
    hdf_file_path: SignalRW[str]
    ...
```
This would then be filled in in the `PandA` device, then passed to the `PandaHDFWriter.__init__(self, data: DataBlock, ...)`

(At the moment the pvi names don't have underscores where they should after https://github.com/PandABlocks/PandABlocks-ioc/issues/101)

Reply:
Yeah, I was planning on putting this into `panda.py` after Eva's change which changes that file. It might not conflict though, I can add it here

Reply:
The key thing is, this is a request for the pvi code to fill it in, rather than a dataclass where the already filled in signals are passed to it